### PR TITLE
Fix `shape` intrinsic for scalar arg

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fixes: https://github.com/lfortran/lfortran/issues/3081

### CODE REVIEW

File: src/libasr/pass/intrinsic_array_function_registry.h [hunk 1]
```diff
@@ -764,15 +764,19 @@ namespace Shape {
...
         Vec<ASR::expr_t *> m_shapes; m_shapes.reserve(al, n_dims);
-        for (size_t i = 0; i < n_dims; i++) {
-            if (m_dims[i].m_length) {
-                ASR::expr_t *e = nullptr;
-                if (extract_kind_from_ttype_t(type) != 4) {
-                    e = i2i(m_dims[i].m_length, extract_type(type));
-                } else {
-                    e = m_dims[i].m_length;
+        if( n_dims == 0 ){
+            m_shapes.push_back(al, i32(0));
```
[the remark this answers is not in the document]
Comment (HarshitaKalani) [reply src/libasr/pass/intrinsic_array_function_registry.h]: Does this look better now?

File: integration_tests/intrinsics_105.f90 [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  INTEGER, DIMENSION(-1:1, -1:2, -1:10) :: A
+  INTEGER :: result(3)
+
+  result = shape(A)
+  print*, shape(A)
+  
+  if ( result(1) - 3 /= 0 ) error stop
+  if ( result(2) - 4 /= 0 ) error stop
+  if ( result(3) - 12 /= 0 ) error stop
+  if ( size(shape(42)) - 0 /= 0 ) error stop
```
Comment (certik) [inline integration_tests/intrinsics_105.f90:11]: Integers can be compared directly:
```suggestion
  if ( result(1) /= 3 ) error stop
  if ( result(2) /= 4 ) error stop
  if ( result(3) /= 12 ) error stop
  if ( size(shape(42)) /= 0 ) error stop
```

File: src/libasr/pass/intrinsic_array_function_registry.h [hunk 3]
```diff
@@ -803,7 +808,7 @@ namespace Shape {
         ASR::ttype_t *return_type = b.Array({n_dims},
             TYPE(ASR::make_Integer_t(al, loc, kind)));
         ASR::expr_t *m_value = eval_Shape(al, loc, return_type, args);
-
+
```
Comment (certik) [inline src/libasr/pass/intrinsic_array_function_registry.h:811]: Remove trailing whitespace.

**Review by certik — APPROVED** [review]
I think that's fine, left some minor comments.

**Review by certik — APPROVED** [review]

**Pull request merged**